h2o-py/h2o/utils/shared_utils.py

```diff
@@ -49,6 +49,37 @@ class List(list):
     pass
 
 
+class LookupList(list):
```

Comment:
Not sure I understand your point here: what would be an inappropriate use of this data structure? It's just a list. Exposing the `set` is just a small optimization, the user could simply do `set(my_list)` as usual: would work, just slightly inefficient in this case.
Looks like you're overcomplicating this.

Reply:
My point was that you can use it the wrong way and have n^2 complexity because you don't think about that at that point. There is nothing that guides you not to do it this way.

You can just treat it as a list and we would be where we started.

Reply:
> You can just treat it as a list and we would be where we started.

Give me an example how you can treat it as a list with poor perf, because I can't see any.
The structure is designed in such a way that if you're doing the typically slow operation on lists, `elem in my_list`, then this will be fast.
I mean like any other datastructure you pick it for what it's good at: you want random access, pick list or tuple; want fast lookup or set operations, pick a set; want random access+fast lookup, pick this LookupSeq.
By exposing a structure that don't make any choice but exposes `as_list`, `as_set`, you're just postponing this decision:
```
names = yourSeq(a_lot_of_names)
...
xzy = names.as_list()
...
for (c in range(1000000)):
    if c in xzy:
         print("BAM!")
```

Reply:
maybe I confused you by exposing this `set` method on the structure: this method is not necessary, I can remove it. It is just slightly faster to do:
```python
names = LookupSeq(...)

x = names.set() - {y}
```
than
```python
names = LookupSeq(...)
 x = set(names) - {y}
 ```
 
 but both work, and the goal of `LookupSeq` is not to expose this `set()` method, it is to use the set internally for lookups only.

Reply:
I think one example could `list(LookupList)` or any filtering that will produce an iterator. The moment you have to type as_list should trigger something in you, if it doesn't there is no help (that is why this needs a test, but a test covers existing scenarios).

Of course, my proposal won't protect you from doing what you demonstrated. You can still do it wrong. I just thought we could use it to guide users (maybe not all, maybe some). 

Personally, I don't see anything wrong with having a structure that just contains the same data with 2 views and you as a user need to choose what view you want. But I can also see your side.

Reply:
It is in the end up to you to decide - I am only suggesting.

Reply:
I'm not convinced, so I'll leave it as it is :)

will come up with a test for the original issue, the current unit test was trying to replicate this at at the structure level, but just to get a proper confirmation.

for `ignored_columns`, we're just iterating through it to fill a set, no lookup, so I can't see how it would benefit from `LookupSeq`. Whatever the structure we use for `ignored_columns` (could also be a set), full iteration will always be O(N)

Reply:
for ignore_columns, there is

ignored_columns and ignored_columns_set which is the as what your structure is doing - no speed up there

also, the test for ignored column is meant to catch an issue that would develop over time